sdk/appconfiguration/src/configuration/mod.rs

```diff
@@ -0,0 +1,77 @@
+use azure_core::auth::TokenCredential;
+use futures::{executor::block_on, stream::StreamExt};
+use std::{collections::HashMap, sync::Arc};
+
+#[derive(Clone)]
+pub struct ConfigurationExplorer {
+    client: azure_svc_appconfiguration::Client,
+}
+
+impl ConfigurationExplorer {
+    pub fn new(token_credential: Arc<dyn TokenCredential>) -> Self {
+        let name =
+            std::env::var("AZCONFIG_NAME").expect("Missing AZCONFIG_NAME environment variable.");
+        let endpoint = format!("https://{name}.azconfig.io");
+        let scopes = &["https://azconfig.io"];
+
+        let client = azure_svc_appconfiguration::Client::builder(token_credential)
+            .endpoint(endpoint)
+            .scopes(scopes)
+            .build();
+
+        Self { client }
+    }
+
+    pub async fn get_value_async(&self, key: String) -> String {
+        let value = match self.client.clone().get_key_value(key).await {
+            Ok(rs) => rs.value,
+            Err(err) => {
+                log::debug!("*ERROR :  {:?}", err);
+                None
+            }
+        };
+
+        match value {
+            Some(value) => value,
+            None => String::from(""),
+        }
+    }
+
+    pub fn get_value(&self, key: String) -> String {
+        block_on(self.get_value_async(key))
+    }
+
+    pub async fn get_values_async(&self, label: String) -> HashMap<String, String> {
+        let mut map = HashMap::new();
+        let mut stream = self
+            .client
+            .clone()
+            .get_key_values()
+            .label(label)
+            .into_stream();
+        while let Some(rs) = stream.next().await {
+            match rs {
+                Ok(rs) => {
+                    let items = rs
+                        .items
+                        .iter()
+                        .map(|it| match (&it.key, &it.value) {
+                            (Some(key), Some(value)) => (key.to_string(), value.to_string()),
+                            _ => (String::from(""), String::from("")),
```

Comment:
Can you describe the use case for adding the key & value of empty strings here?

Reply:
good catch. It is not required, I  replaced it to None.